nixos/modules/programs/screen.nix

```diff
@@ -1,33 +1,41 @@
 { config, lib, pkgs, ... }:
 
 let
-  inherit (lib) mkOption mkIf types;
   cfg = config.programs.screen;
 in
 
 {
-  ###### interface
-
   options = {
     programs.screen = {
+      enable = lib.mkEnableOption (lib.mdDoc "screen, a basic terminal multiplexer");
+
+      package = lib.mkPackageOptionMD pkgs "screen" { };
 
-      screenrc = mkOption {
-        default = "";
-        description = lib.mdDoc ''
-          The contents of /etc/screenrc file.
+      screenrc = lib.mkOption {
+        type = with lib.types; nullOr lines;
+        example = ''
+          defscrollback 10000
+          startup_message off
         '';
-        type = types.lines;
+        description = lib.mdDoc "The contents of {file}`/etc/screenrc` file";
       };
     };
   };
 
-  ###### implementation
-
-  config = mkIf (cfg.screenrc != "") {
-    environment.etc.screenrc.text = cfg.screenrc;
-
-    environment.systemPackages = [ pkgs.screen ];
+  config = {
+    # TODO: Added in 24.05, remove before 24.11
+    assertions = [
+      {
+        assertion = cfg.screenrc != null -> cfg.enable;
+        message = "`programs.screen.screenrc` has been configured, but `programs.screen.enable` is not true";
+      }
```

Comment:
This breaks a core assumption in the nixos modules. Many big modules eg. plasma5 set config options for other modules that are only used when they are activated.

Reply:
Sure, that makes sense.

I grepped through nixpkgs and only found one reference to screenrc made in the weechat docs though, so I don't think it should be a big deal. I think it is important that the user is notified that the screenrc they configured in a previous nixos version isn't going to realise without enabling the module.

Would you be fine with switching out `assertions` for `warnings`?

Reply:
Switching it to a warning would allow people to disable screen while not being forced to clear their screenrc setting. I think that's better.